### PR TITLE
feature(ui-ux): verify smart contract

### PR DESCRIPTION
**Pull request by fullstackninja864**
#### Which issue(s) does this PR fixes?:

Fixes #

#### Developer Checklist:

- [ ] Read your code changes at least once
- [ ] Your UI implementation visually matched the rendered design\*
- [ ] Unit tests\*
- [ ] Added e2e tests\*

### CODE REVIEW

File: src/pages/contract/verify/index.tsx [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+                )}
+                <span className="text-white-50 ml-2 mr-1">I agree to the</span>
+              </button>
+              <a
+                target="_blank"
+                data-testid="terms-of-service"
+                href="/contract/verify/terms"
+                className="text-lightBlue brand-gradient-1 active:brand-gradient-2 bg-clip-text hover:text-transparent transition-all ease-in duration-300"
+              >
+                terms of service
```
Comment (pierregee) [inline src/pages/contract/verify/index.tsx]:
```suggestion
                Terms of Service
```

File: src/pages/contract/verify/terms.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+          <div className="text-base md:text-lg -tracking-[0.02em] text-white-50">
+            If you are registering to use the Services on behalf of a legal
+            entity, you further represent and warrant that (i) such legal entity
+            is duly organized and validly existing under the applicable laws of
+            the jurisdiction of its organization, and (ii) you are duly
+            authorized by such legal entity to act on its behalf.
+          </div>
+        </div>
+        <div className="mt-6">
+          <div className="text-base md:text-lg -tracking-[0.02em] text-white-50">
```
Comment (pierregee) [inline src/pages/contract/verify/terms.tsx]: minor nit: can put all the repetitive class in a variable 🧹

File: next.config.js [hunk 3]
```diff
@@ -15,7 +15,7 @@ const securityHeaders = [
       };` +
       `style-src 'self' fonts.googleapis.com 'unsafe-inline';` +
       `font-src fonts.gstatic.com;` +
-      `connect-src 'self' blockscout-index.changi.dfi.team ${
+      `connect-src 'self' blockscout-index.changi.dfi.team solc-bin.ethereum.org raw.githubusercontent.com ${
```
Comment (pierregee) [inline next.config.js]: Since we aren't gonna be needing any other api calls from these 2 sites, is it possible to use the specific url here?

**Review by pierregee — CHANGES_REQUESTED** [review]

File: next.config.js (hunk 3 above)
Comment (fullstackninja864) [reply next.config.js]: Will add that specific path

File: src/pages/contract/verify/_components/StepOne.tsx [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+import SmartContractApi from "@api/SmartContractApi";
+import { SCVersionsBuilds } from "@api/types";
+import InputComponent from "@components/commons/InputComponent";
+
+enum ContractLanguage {
+  Solidity = "Solidity",
+  Vyper = "Vyper",
+}
+
+export default function StepOne({ onSubmit, defaultDropdownValue }) {
```
Comment (pierregee) [inline src/pages/contract/verify/_components/StepOne.tsx]: let's add types

File: src/api/SmartContractApi.ts [hunk 5]
```diff
@@ -24,6 +27,28 @@ export default {
...
+  getVyperVersions: async (): Promise<SCVersionsResponseProps> => {
+    const res = await fetch(VYPER_VERSIONS_URL);
+    return wrapResponse<SCVersionsResponseProps>(res);
+  },
+  verifySmartContract: async (
+    network: NetworkConnection,
+    data
+  ): Promise<any> => {
+    const baseUrl = getBaseUrl(network);
+    const res = await fetch(`${baseUrl}api?module=contract&action=verify`, {
```
Comment (lykalabrada) [inline src/api/SmartContractApi.ts]:
```suggestion
    const res = await fetch(`${baseUrl}/api?module=contract&action=verify`, {
```

File: next.config.js [hunk 6]
```diff
@@ -15,7 +15,7 @@ const securityHeaders = [
       };` +
       `style-src 'self' fonts.googleapis.com 'unsafe-inline';` +
       `font-src fonts.gstatic.com;` +
-      `connect-src 'self' blockscout-index.changi.dfi.team ${
+      `connect-src 'self' blockscout-index.changi.dfi.team solc-bin.ethereum.org/linux-amd64/list.json raw.githubusercontent.com/blockscout/solc-bin/main/vyper.list.json blockscout.com 35.197.159.154:4000 ${
```
Comment (lykalabrada) [inline next.config.js]: We can replace `35.197.159.154:4000` with `blockscout-index.changi.dfi.team` right?
```suggestion
      `connect-src 'self' blockscout-index.changi.dfi.team solc-bin.ethereum.org/linux-amd64/list.json raw.githubusercontent.com/blockscout/solc-bin/main/vyper.list.json blockscout.com ${
```

**Review by lykalabrada — CHANGES_REQUESTED** [review]

**Review by lykalabrada — APPROVED** [review]

**Review by lykalabrada — APPROVED** [review]

**Review by pierregee — APPROVED** [review]

**Pull request merged**